src/modules/transaction/schemas.rs

```diff
@@ -0,0 +1,158 @@
+use crate::types;
+use paperclip::actix::Apiv2Schema;
+use validator::Validate;
+
+// *** Requests ***
+
+#[derive(
+    Validate, Debug, Clone, PartialEq, Eq, serde::Serialize, serde::Deserialize, Apiv2Schema,
+)]
+pub struct TransactionRequest {
+    #[validate(custom = "crate::errors::validate_crypto_hash")]
+    pub transaction_hash: Option<types::TransactionHash>,
+    pub receipt_id: Option<types::ReceiptId>,
+}
+
+#[derive(
+    Validate, Debug, Clone, PartialEq, Eq, serde::Serialize, serde::Deserialize, Apiv2Schema,
+)]
+pub struct TransactionsRequest {
+    #[validate(custom = "crate::errors::validate_account_id")]
+    pub account_id: types::AccountId,
+    #[validate(custom = "crate::errors::validate_account_id")]
+    pub contract_id: Option<types::AccountId>,
+}
+
+#[derive(
+    Validate, Debug, Clone, PartialEq, Eq, serde::Serialize, serde::Deserialize, Apiv2Schema,
+)]
+pub struct ReceiptsRequest {
+    #[validate(custom = "crate::errors::validate_crypto_hash")]
+    pub transaction_hash: types::TransactionHash,
+}
+
+#[derive(
+    Validate, Debug, Clone, PartialEq, Eq, serde::Serialize, serde::Deserialize, Apiv2Schema,
+)]
+pub struct ActionReceiptsRequest {
+    #[validate(custom = "crate::errors::validate_account_id")]
+    pub account_id: types::AccountId,
+    #[validate(custom = "crate::errors::validate_account_id")]
+    pub contract_id: Option<types::AccountId>,
+}
+
+// *** Responses ***
+#[derive(Debug, Clone, PartialEq, Eq, serde::Serialize, serde::Deserialize, Apiv2Schema)]
+pub struct TransactionResponse {
+    pub transaction: Transaction,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, serde::Serialize, serde::Deserialize, Apiv2Schema)]
+
+pub struct TransactionsResponse {
+    pub transactions: Vec<Transaction>,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, serde::Serialize, serde::Deserialize, Apiv2Schema)]
+
+pub struct ReceiptsResponse {
+    pub receipts: Vec<Receipt>,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, serde::Serialize, serde::Deserialize, Apiv2Schema)]
+pub struct ActionReceiptsResponse {
+    pub action_receipts: Vec<ActionReceipt>,
+}
+
+// *** Types ***
+
+#[derive(Debug, Clone, PartialEq, Eq, serde::Serialize, serde::Deserialize, Apiv2Schema)]
+pub struct Transaction {
```

Comment:
https://github.com/near/near-explorer/blob/40253edaccca7514819916b16d4368d2ac10ec01/backend/src/router/transaction/get.ts#L301-L312


Transaction definition for current near-explorer frontend

Reply:
Good point and I agree with you! I intentionally changed them to be all String types so I can quickly return dummy values from the handlers for the time being. Will try to make as much use of enums and types for my next review :) 👍